internal/backend/workloadproxy/reconcile_utils.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2025 Sidero Labs, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+package workloadproxy
+
+import (
+	"fmt"
+	"iter"
+	"slices"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	xmaps "github.com/siderolabs/gen/maps"
+	"github.com/siderolabs/gen/pair"
+)
+
+// aliasToCluster is a data structure that maps aliases to clusters and ports. It doesn't keep tarck of active probes,
+// but it keeps track of the in-use port for the said probe for each cluster.
+type aliasToCluster struct {
+	aliasToPort map[alias]pair.Pair[*clusterData, port]
+	slc         uniqueSlice[*clusterData]
```

Comment:
can we do something more simple here? 

this is so many layers of abstraction, that imho it makes it virtually impossible to understand and read.

my 2 cents

Reply:
This is actually my Nth attempt of refactoring this. I don't have any better solutions here, since we need a fast access for both cluster and aliases. You could store every cluster inside the slice and lookup aliases using linear search and accessing aliases inside clusterData, but that will look even worse and will add O(N) complexity to all alias lookups.

I didn't want to leave only `aliasToPort` map, because that way, when you need to update/remove cluster you have to walk entire map.

Reply:
I'm actually going crazy about it. `uniqueSlice` is inefficient as it requires full iteration all the time. 

This is yet another layer of abstraction, specifically bad container implementation just for sake of what?

aliasToCluster is what? `map[string]string`, that is `alias -> (clusterID, port)`, and that's it? 

and a separate `map[string]*Cluster` which is `alias -> Cluster`? 

if we need more lookups for whatever reason, we can also have `cluster -> []alias`, like `map[string]map[string]struct{}` ? 

but something manageable without `pair` with ugly F1/F2, maps instead of bad unique slice container?

all it has to do is to act on reconcile events, nothing more, nothing less.

Reply:
Okay, lets try different approach:
- Replaced `uniqueSlice` with `map[resource.ID]*clusterData`.
- Replaced `pair.Pair` with the actual struct with proper names.
- Removed `uniqueSlice` type and tests.

Does it look better?